### PR TITLE
add missing LOG_ASYNC_ENABLED definition

### PR DESCRIPTION
**Pull request by haifengkao**
it is required by CocoaLumberjack

### CODE REVIEW

**Comment by swisspol** [comment]
I couldn't find any information about this define being required by cocoalumberjack. What happens if it's not defined?

File: GCDWebServer/Core/GCDWebServerPrivate.h [hunk 1]
```diff
@@ -92,6 +92,13 @@
 
 #define __GCDWEBSERVER_LOGGING_FACILITY_COCOALUMBERJACK__
 
+/**
+ * Whether async should be used by log messages, excluding error messages that are always sent sync.
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerPrivate.h]: Please remove this comment which is not needed inside gcdwebserver’s private header.

**Comment by haifengkao** [comment]
See https://github.com/CocoaLumberjack/CocoaLumberjack/issues/778
For some reason, when GCGWebServer is linked as a dynamic framework, the line `#import "DDLogMacros.h"` in `CocoaLumberjack.h` doesn't take any effects. Therefore, `LOG_ASYNC_ENABLED` is undefined. 

I guess it has something to do with the module preprocessing rules. But I don't really understand the underlying mechanism.

**Comment by haifengkao** [comment]
any updates?

**Comment by swisspol** [comment]
Does this problem still happen with very latest GCDWebServer and CocoaLumberJack? If so can you provide a simple test project or instructions to repro?

**Comment by haifengkao** [comment]
CocoaLumberjack 3 doesn't have this issue anymore

**Pull request closed without merging**